Add Parabricks DeepVariant module

HudoGriz opened the pull request:
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test_singleton,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

## Description
This PR adds the `parabricks/deepvariant` module for GPU-accelerated variant calling in the `nf-core/raredisease` pipeline.

At the **Clinical Institute of Special Laboratory Diagnostics, University Medical Centre Ljubljana**, we have been using this implementation internally, as it significantly accelerates variant calling without introducing major differences in benchmark results compared to the CPU-based DeepVariant implementation.

We are contributing this module so that:
- Other users can benefit from GPU acceleration in their rare disease analyses.
- Our version control and maintenance can be unified within the nf-core framework.

## Tests performed
We successfully ran the following tests:
```
nextflow run . -profile test,singularity,gpu --variant_caller parabricks_deepvariant --outdir test_results

nextflow run . -profile test_singleton,singularity,gpu --variant_caller parabricks_deepvariant --outdir test_results_singleton
```

We are happy to receive feedback on how this contribution can be better integrated and aligned with nf-core standards, as well as any recommendations for improving its adoption and usability across the wider community.

HudoGriz committed:
Add Parabricks DeepVariant module

ramprasadn reviewed (commented):
Thanks for this PR @HudoGriz! Could you add a changelog entry for this PR?

HudoGriz commented:
Rebased to the latest `dev` and updated `CHANGELOG.md`.
Any insights on the new test error would be appreciated:
```
     Failed to open sentieon license file : No such file or directory
    cmdline: /opt/conda/share/sentieon-202503-0/libexec/util sort -r genome.fasta -t 4 -o test_sorted.bam --sam2bam -
    Failed to open sentieon license file : No such file or directory
```

ramprasadn committed:
Merge branch 'dev' into add-parabricks-deepvariant

HudoGriz commented:
Ah I see the issue, fork PRs don't have access to repository secrets, which causes the Failed to open sentieon license file : No such file or directory.

A maintainer could push my branch to the main repository (nf-core/raredisease:add-parabricks-deepvariant) so the CI can access the Sentieon license secrets, an then the tests should complete. Alternatively, if you prefer, I can close this PR and you can create a new one from a branch in the main repo.
